### DISCUSSION TITLE
how to select one GPU that has more memory with ID?

### DISCUSSION THREAD

**Discussion by ning-2022** [facebookresearch/esm#352] [Q&A]
Hi, everyone,

     I have a problem that ESMfold had reported the cuda out of memory. My machine has two GPU, one is K2000, a very old GPU card, with 1GB memory, another one is RTX2080Ti, with 12GB memory. My question is how to choose the GPU that be used in protein structure prediction, because of that ESMfold automatically used the K2000 GPU. Any suggestions are welcome!

Best,
Ning

**Comment by tomsercu** [comment]
before invoking python you can set an environment variable to make only one gpu visible: `export CUDA_VISIBLE_DEVICES=1`. [LINK](https://stackoverflow.com/questions/39649102/how-do-i-select-which-gpu-to-run-a-job-on)
Alternatively you can use pytorch to do this, use something like `model.to(device="cuda:1")` [LINK](https://pytorch.org/docs/stable/notes/cuda.html)

**Reply by ning-2022** [reply]
Thanks for your suggestions. Yes, it works.
